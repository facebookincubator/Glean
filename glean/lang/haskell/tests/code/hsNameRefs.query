query: hs.NameRefs _
transform: [normord, []]
