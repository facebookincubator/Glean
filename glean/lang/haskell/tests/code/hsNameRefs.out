[
  "@generated",
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "A", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 279 } },
        { "kind": 2, "span": { "length": 1, "start": 442 } },
        { "kind": 2, "span": { "length": 1, "start": 579 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "C", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 305 } },
        { "kind": 2, "span": { "length": 1, "start": 543 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "C2", "namespace_": 1 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 2, "start": 647 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "C3", "namespace_": 1 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 2, "start": 623 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "R", "namespace_": 1 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 1, "start": 609 } },
        { "kind": 2, "span": { "length": 2, "start": 467 } },
        { "kind": 2, "span": { "length": 2, "start": 479 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "R", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 295 } },
        { "kind": 2, "span": { "length": 1, "start": 596 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "T", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 285 } },
        { "kind": 2, "span": { "length": 1, "start": 603 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "a", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 273 } },
        { "kind": 2, "span": { "length": 1, "start": 370 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "a", "namespace_": 2 } },
          "sort": { "internal": { "length": 1, "start": 424 } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 449 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "a", "namespace_": 2 } },
          "sort": { "internal": { "length": 1, "start": 459 } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 486 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "a", "namespace_": 2 } },
          "sort": { "internal": { "length": 1, "start": 508 } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 1, "start": 501 } },
        { "kind": 2, "span": { "length": 1, "start": 523 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "a", "namespace_": 2 } },
          "sort": { "internal": { "length": 15, "start": 591 } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 1, "start": 598 } },
        { "kind": 2, "span": { "length": 1, "start": 605 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "f", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 324 } },
        { "kind": 2, "span": { "length": 1, "start": 591 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "f1", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 2, "start": 467 } },
        { "kind": 2, "span": { "length": 2, "start": 656 } },
        { "kind": 2, "span": { "length": 2, "start": 663 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "f2", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 2, "start": 479 } },
        { "kind": 2, "span": { "length": 2, "start": 611 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "m", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 1, "start": 557 } },
        { "kind": 2, "span": { "length": 1, "start": 634 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "r", "namespace_": 0 } },
          "sort": { "internal": { "length": 1, "start": 630 } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 659 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "x", "namespace_": 0 } },
          "sort": { "internal": { "length": 1, "start": 559 } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 1, "start": 563 } },
        { "kind": 2, "span": { "length": 1, "start": 568 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "x", "namespace_": 0 } },
          "sort": { "internal": { "length": 1, "start": 617 } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 626 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "zero", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 4, "start": 315 } },
        { "kind": 2, "span": { "length": 4, "start": 571 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.Base" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "String", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 6, "start": 375 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.Base" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "map", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 3, "start": 386 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.Base" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "ord", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 0, "span": { "length": 3, "start": 364 } },
        { "kind": 2, "span": { "length": 3, "start": 651 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Classes" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "==", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 2, "start": 565 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Classes" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Eq", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 2, "start": 498 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Types" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Bool", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 4, "start": 528 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Types" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Char", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 4, "start": 473 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Types" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Float", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 5, "start": 431 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Types" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Int", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 2, "span": { "length": 3, "start": 412 } },
        { "kind": 2, "span": { "length": 3, "start": 545 } },
        { "kind": 2, "span": { "length": 3, "start": 640 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/A.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Unicode" },
              "unit": { "key": "base" }
            }
          },
          "occ": { "key": { "name": "toLower", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 0, "span": { "length": 7, "start": 355 } },
        { "kind": 2, "span": { "length": 7, "start": 390 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "R", "namespace_": 1 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 3, "start": 331 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "R", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 320 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "a", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 312 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "f1", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 2, "start": 337 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "A" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "f2", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 2, "start": 347 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "B" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "b", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 228 } },
        { "kind": 2, "span": { "length": 1, "start": 279 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "B" }, "unit": { "key": "main" } }
          },
          "occ": { "key": { "name": "r", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 1, "span": { "length": 1, "start": 234 } },
        { "kind": 2, "span": { "length": 1, "start": 315 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.Base" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "$", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 1, "start": 302 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.Base" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "map", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 3, "start": 304 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.Base" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "ord", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [
        { "kind": 0, "span": { "length": 3, "start": 273 } },
        { "kind": 2, "span": { "length": 3, "start": 308 } }
      ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": { "name": { "key": "GHC.List" }, "unit": { "key": "base" } }
          },
          "occ": { "key": { "name": "reverse", "namespace_": 0 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 7, "start": 294 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Types" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Char", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 4, "start": 322 } } ]
    }
  },
  {
    "key": {
      "file": { "key": "glean/lang/haskell/tests/code/B.hs" },
      "target": {
        "key": {
          "mod": {
            "key": {
              "name": { "key": "GHC.Types" },
              "unit": { "key": "ghc-prim" }
            }
          },
          "occ": { "key": { "name": "Int", "namespace_": 3 } },
          "sort": { "external": { } }
        }
      },
      "uses": [ { "kind": 2, "span": { "length": 3, "start": 285 } } ]
    }
  }
]